wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -274,9 +280,16 @@ def _entry_from_obj(
         )
 
     @staticmethod
-    def _etag_from_obj(obj: Union["boto3.s3.Object", "boto3.s3.ObjectSummary"]) -> ETag:
+    def _etag_from_obj(
+        obj: Union[
+            "boto3.s3.Object", "boto3.s3.ObjectVersion", "boto3.s3.ObjectSummary"
+        ]
+    ) -> ETag:
         etag: ETag
-        etag = obj.e_tag[1:-1]  # escape leading and trailing quote
+        if hasattr(obj, "load"):
+            etag = obj.e_tag[1:-1]  # escape leading and trailing quote
```

Comment:
nit, why not just check for the `e_tag` attr directly rather then inferring frmo the type of object. Also, if infering from the type, would be cleaning to read to check the type directly

Reply:
I decided not to infer from the type since an `ObjectVersion` read from the list/collection returned by `object_versions()` has the e_tag attribute, while `ObjectVersion` returned by `bucket.ObjectVersion` doesn't